### PR TITLE
Go module support for busybox

### PR DESCRIPTION
**Pull request by hugelgupf**
This adds Go module support for busybox, while preserving support for GOPATH and vendoring.

I will split this into its own repository soon, so we can also have the bazel/blaze scripts there, without having to support them for all of u-root.

### CODE REVIEW

**Comment by JulienVdG** [comment]
I tested this by cloning, pointing GOPATH to some new directory, creating a go.mod with `go mod init github.com/u-root/u-root`, I had to drop a dependency to build, go complained about some v0 versus v3 for a package...
then I could build u-root and use the local u-root for building an initramfs !

I also read through the diff and I think I understand most of the changes and didn't spot anything that looked wrong !

I did a last non-regression quick test, for the regular GOPATH case I did checkout this PR in my test environment and run the tastevin linuxboot test on an actual winterfell: all tests passed

I'm no expert but LGTM

**Comment by sevki** [comment]
Hi folks :wave: first time contributor here. 
This PR seems to re-implement a lot of what [`golang.org/x/tools/go/packages` ](https://godoc.org/golang.org/x/tools/go/packages)

Sorry, I do not have a lot of context, is not using the `packages` pkg for historical/backwards-compat reasons?

**Comment by hugelgupf** [comment]
Hey @sevki, I built this a few months ago but haven't updated it since. It does use go/packages. Where should I be using it that we're not? It's possible go/packages has gained functionality it didn't have a few months ago in the meantime, but I'm just guessing (haven't looked yet, will soon)

File: pkg/golang/build.go [hunk 1]
```diff
@@ -79,23 +91,70 @@ func (c Environ) Version() (string, error) {
 	return s[2], nil
 }
 
-// Deps lists all dependencies of the package given by `importPath`.
-func (c Environ) Deps(importPath string) (*ListPackage, error) {
+// Find lists all dependencies of the package given by `importPath`.
+func (c Environ) Find(pattern string) ([]*Package, error) {
```
Comment (sevki) [inline pkg/golang/build.go:95]: I think you can replace this line :)

```suggestion
func (c Environ) Find(pattern string) ([]*packages.Package, error) {return packages.Load(c.Config, pattern)}
```

File: pkg/golang/build.go [hunk 2]
```diff
@@ -6,58 +6,70 @@
...
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
+	"time"
 )
 
 type Environ struct {
 	build.Context
+
```
Comment (sevki) [inline pkg/golang/build.go:22]: add this here
```suggestion

	*packages.Config
```

File: pkg/golang/build.go [hunk 3]
```diff
@@ -6,58 +6,70 @@
...
-
-// PackageByPath retrieves information about a package by its file system path.
-//
-// `path` is assumed to be the directory containing the package.
-func (c Environ) PackageByPath(path string) (*build.Package, error) {
-	abs, err := filepath.Abs(path)
-	if err != nil {
-		return nil, err
+	return Environ{
+		Context:     build.Default,
```
Comment (sevki) [inline pkg/golang/build.go:30]:
```suggestion
 		Context:     build.Default,
		Config: &packages.Config{
			Mode:       packages.LoadSyntax,
		},
```

**Pull request closed without merging**